tests/src/test/java/alluxio/client/cli/job/DistributedCmdTest.java

```diff
@@ -0,0 +1,145 @@
+    mNoopPlanConfig = Mockito.mock(NoopPlanConfig.class);
+    mFileSystem = Mockito.mock(FileSystem.class);
+    mRetryPolicy = new CountingRetry(5);
+
+    mLoadURIStatus = Mockito.mock(URIStatus.class);
+    mMigrateURIStatus = Mockito.mock(URIStatus.class);
+    mPersistURIStatus = Mockito.mock(URIStatus.class);
+    mNoopPlanURIStatus = Mockito.mock(URIStatus.class);
+  }
+
+  private void initialize() throws Exception {
+    Mockito.when(mLoadURIStatus.getLength()).thenReturn(LOAD_FILE_LENGTH);
+
+    Mockito.when(mLoadJobConfig.getFilePath()).thenReturn(LOAD);
+    Mockito.when(mMigrateJobConfig.getSource()).thenReturn(MIGRATE);
+    Mockito.when(mPersistJobConfig.getFilePath()).thenReturn(PERSIST);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(LOAD))).thenReturn(mLoadURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(LOAD)).getLength())
+            .thenReturn(LOAD_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(MIGRATE))).thenReturn(mMigrateURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(MIGRATE)).getLength())
+            .thenReturn(MIGRATE_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(PERSIST))).thenReturn(mPersistURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(PERSIST)).getLength())
+            .thenReturn(PERSIST_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(NOOP_PLAN))).thenReturn(mNoopPlanURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(NOOP_PLAN)).getLength())
+            .thenReturn(NOOP_PLAN_FILE_LENGTH);
+  }
+
+  @Test
+  public void testSupportedConfigForCompleteStatus() throws Exception {
+    initialize();
+    //Run for supported configs.
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mLoadJobConfig, mFileSystem, mRetryPolicy);
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mMigrateJobConfig, mFileSystem, mRetryPolicy);
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mPersistJobConfig, mFileSystem, mRetryPolicy);
+
+    ////should expect function calls to fileSystem.getStatus(new AlluxioURI(filePath)).getLength(),
+    // and cooresponding file sizes obtained.
+    double loadFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_JOB_DISTRIBUTED_LOAD_FILE_SIZE.getName()).getValue();
+    double migrateFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_MIGRATE_JOB_FILE_SIZE.getName()).getValue();
+    double persistFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_ASYNC_PERSIST_FILE_SIZE.getName()).getValue();
+
+    assertEquals(loadFileSize, LOAD_FILE_LENGTH, 0);
+    assertEquals(migrateFileSize, MIGRATE_FILE_LENGTH, 0);
+    assertEquals(persistFileSize, PERSIST_FILE_LENGTH, 0);
+  }
+
+  @Test
+  public void testUnSupportedConfigForCompleteStatus() throws Exception {
+    MetricsSystem.resetAllMetrics();
```

Comment:
why does this test need to reset metrics but the previous one does not? similarly, why does the previous test need to initialize but this one does not?

does it make more sense to add the reset() and initialize() in @Before to run before each test? this makes the test more robust

Reply:
If it doesn't have the reset, the metrics would not be 0s before the test. Good point, will move the reset to @before.
Maybe i need to rename the function `initialize`, it just mocks the class methods. I can move that part to @before as well.